integration_tests/framework/runner_container.go

```diff
@@ -0,0 +1,314 @@
+
+		executor: cliWrappers.NewCliExecutor(),
+		name:     name,
+		image:    image,
+		env:      make(map[string]string),
+		volumes:  make(map[string]string),
+		ports:    make(map[string]string),
+		results:  make(map[string]string),
+	}
+}
+
+// NewBuildCliRunnerContainer creates NewTestRunnerContainer
+// with additional settings for running the Build CLI.
+func NewBuildCliRunnerContainer(name, image string) *TestRunnerContainer {
+	container := NewTestRunnerContainer(name, image)
+
+	container.AddVolume(GetCliBinPath(), path.Join("/usr/bin/", KonfluxBuildCli))
+	container.AddNetwork("host")
+	if Debug {
+		container.AddPort("2345", "2345")
+	}
+	container.AddEnv("KBC_LOG_LEVEL", "debug")
+
+	return container
+}
+
+func (c *TestRunnerContainer) ensureContainerNotStarted() {
+	if c.containerStatus != ContainerStatus_NotStarted {
+		panic("the operation can be done only before the container is started")
+	}
+}
+
+func (c *TestRunnerContainer) ensureContainerRunning() {
+	if c.containerStatus != ContainerStatus_Running {
+		panic("the operation can be done only on running container")
+	}
+}
+
+func (c *TestRunnerContainer) SetWorkdir(workdir string) {
+	c.ensureContainerNotStarted()
+	c.workdir = workdir
+}
+
+func (c *TestRunnerContainer) AddEnv(key, value string) {
+	c.ensureContainerNotStarted()
+	c.env[key] = value
+}
+
+func (c *TestRunnerContainer) AddVolume(hostPath, containerPath string) {
+	c.ensureContainerNotStarted()
+	c.volumes[hostPath] = containerPath
+}
+
+func (c *TestRunnerContainer) AddPort(hostPort, containerPort string) {
+	c.ensureContainerNotStarted()
+	c.ports[hostPort] = containerPort
+}
+
+func (c *TestRunnerContainer) AddNetwork(networkName string) {
+	c.ensureContainerNotStarted()
+	c.networks = append(c.networks, networkName)
+}
+
+// ContainerExists checks for container with the same name.
+func (c *TestRunnerContainer) ContainerExists(isRunning bool) (bool, error) {
+	args := []string{"ps", "-q"}
+	if !isRunning {
+		args = append(args, "-a")
+	}
+	args = append(args, "-f", "name="+c.name)
+
+	stdout, stderr, _, err := c.executor.Execute(containerTool, args...)
+	if err != nil {
+		l.Logger.Infof("[stdout]:\n%s\n", stdout)
+		l.Logger.Infof("[stderr]:\n%s\n", stderr)
+		return false, err
+	}
+	return len(stdout) > 0, nil
+}
+
+func (c *TestRunnerContainer) checkContainer() error {
+	existRunning, err := c.ContainerExists(true)
+	if err != nil {
+		return err
+	}
+	if existRunning {
+		return c.Delete()
+	}
+
+	existStopped, err := c.ContainerExists(false)
+	if err != nil {
+		return err
+	}
+	if existStopped {
+		return c.Delete()
+	}
+	return nil
+}
+
+func (c *TestRunnerContainer) Start() error {
+	if err := c.checkContainer(); err != nil {
+		return err
+	}
+
+	args := []string{"run", "--detach", "--name", c.name}
+	for name, value := range c.env {
+		args = append(args, "-e", name+"="+value)
+	}
+	for hostPath, containerPath := range c.volumes {
+		args = append(args, "-v", hostPath+":"+containerPath)
```

Comment:
I've run into some issues. First, the Zot registry failed to start right away:

```
$ podman logs zot-registry 
{"time":"2025-12-15T08:55:18.553095453Z","level":"error","message":"failed to read configuration","error":"open /etc/zot/config.json: permission denied","path":"/etc/zot/config.json","caller":"zotregistry.dev/zot/v2/pkg/cli/server/root.go:936","func":"zotregistry.dev/zot/v2/pkg/cli/server.LoadConfiguration","goroutine":1}
Error: open /etc/zot/config.json: permission denied
Usage:
  zot serve <config> [flags]

Aliases:
  serve, serve

Flags:
  -h, --help   help for serve
```

On my system, that can be solved by adding SELinux flags to the volume mounts:

```suggestion
		args = append(args, "-v", hostPath+":"+containerPath+":z")
```

But this might have to be conditional, depending on whether the target system has SELinux enabled or not (`getenforce` command returns `"Enforcing"`)

Now I'm running into SSL problems, investigating more